### DISCUSSION TITLE
low_class_penalty or use_truck_route options for buses ?

### DISCUSSION THREAD

**Discussion by blaurent49** [valhalla/valhalla#4939] [Q&A]
Hello,
Are there any options similar to low_class_penalty or use_truck_route options for buses ? They seem only available for trucks. However, we wish to avoid small roads for buses.

In the following example, there is a very tight turn that is nearly impracticable in reality for a bus:
![image](https://github.com/user-attachments/assets/b5baad62-4454-43b6-bf78-305eb92d258a)

How can I avoid this road? Is there another mean?

Thank you in advance for your help.

**Comment by kevinkreiser** [comment]
no there arent but thats a great point. current bus derives from automobile but perhaps bus costing should derive from truck since its typically quite a large, heavy and long vehicle. @dnesbitt61 @gknisely what do you guys think? is bus better derived from truck or is there something else in truck that doesnt make sense for a bus?

**Reply by ianthetechie** [reply]
Hmm interesting. Buses are quite similar in many respects, since there are weight and height limits to consider. But many other things like hazmat and (probably) axle count don't usually apply that I'm aware of 😂 buses are also usually able to use HOV lanes (or alternately, cars can use bus lanes if they meet certain criteria). Not sure if the latter is reflected in the truck profile.

Just my 2 cents off the top of my head not having pulled up the truck costing recently.
